### PR TITLE
Add RequiresAssemblyFilesAttribute

### PR DESCRIPTION
**Pull request by mateoatr**
The new attribute will be used by the ILLink analyzer to know which members are unsafe in the presence of a single-file application and help us produce diagnostics.

### CODE REVIEW

**Review by vitek-karas — APPROVED** [review]

**Review by terrajobst — APPROVED** [review]
LGTM

File: src/libraries/System.Runtime/tests/System/Diagnostics/CodeAnalysis/RequiresAssemblyFilesAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        [Theory]
+        [InlineData("https://dot.net")]
+        [InlineData("")]
+        [InlineData(null)]
+        public void TestSetUrl(string url)
+        {
+            var attr = new RequiresAssemblyFilesAttribute(Url = url);
+
+            Assert.Null(attr.Message);
+            Assert.Equal()
```
Comment (eerhardt) [inline src/libraries/System.Runtime/tests/System/Diagnostics/CodeAnalysis/RequiresAssemblyFilesAttributeTests.cs:40]:
```suggestion
            Assert.Equal(url, attr.Url);
```

File: src/libraries/System.Runtime/tests/System/Diagnostics/CodeAnalysis/RequiresAssemblyFilesAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        [InlineData("", null)]
+        [InlineData(null, "https://dot.net")]
+        [InlineData(null, "")]
+        [InlineData(null, null)]
+        public void TestSetMessageAndUrl(string message, string url)
+        {
+            var attr = new RequiresAssemblyFilesAttribute(Message = message, Url = url);
+
+            Assert.Equal(attr.Message = message);
+            Assert.Equal(attr.Url = url);
```
Comment (eerhardt) [inline src/libraries/System.Runtime/tests/System/Diagnostics/CodeAnalysis/RequiresAssemblyFilesAttributeTests.cs:58]:
```suggestion
            Assert.Equal(message, attr.Message);
            Assert.Equal(ur, attr.Url);
```

**Review by eerhardt — APPROVED** [review]
LGTM.

Side note: when do we plan on applying the attribute to the appropriate APIs?

**Comment by mateoatr** [comment]
> Side note: when do we plan on applying the attribute to the appropriate APIs?

I'd guess as soon as we check-in the analyzer in the SDK; although not sure, @agocke?

**Review by stephentoub — APPROVED** [review]

**Pull request merged**